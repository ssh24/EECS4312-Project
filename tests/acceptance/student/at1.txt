-- This is for testing mainly the correct movement of container in simple scenario
-- it moves container c from phase p1 to p2

-- create the tracker
new_tracker(10, 5) 

-- create two phases p1 and p2
new_phase("p1", "packing", 4, <<glass, metal, plastic, liquid>>) 
new_phase("p2", "crushing", 3, <<glass, metal, plastic>>)

-- add several containers including container c
new_container("c", [glass,   3.0], "p1")
new_container("c2", [plastic,   4.0], "p1")
new_container("c3", [metal,   3.0], "p2")
new_container("c4", [liquid,   3.0], "p1")

-- move container c from phase p1 to phase p2
move_container("c", "p1", "p2")

