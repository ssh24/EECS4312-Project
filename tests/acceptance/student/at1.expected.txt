  state 0 ok
  max_phase_radiation: 0.00, max_container_radiation: 0.00
  phases: pid->name:capacity,count,radiation
  containers: cid->pid->material,radioactivity
->new_tracker(10,5)
  state 1 ok
  max_phase_radiation: 10.00, max_container_radiation: 5.00
  phases: pid->name:capacity,count,radiation
  containers: cid->pid->material,radioactivity
->new_phase("p1","packing",4,<<glass, metal, plastic, liquid>>)
  state 2 ok
  max_phase_radiation: 10.00, max_container_radiation: 5.00
  phases: pid->name:capacity,count,radiation
    p1->packing:4,0,0.00,{glass,metal,plastic,liquid}
  containers: cid->pid->material,radioactivity
->new_phase("p2","crushing",3,<<glass, metal, plastic>>)
  state 3 ok
  max_phase_radiation: 10.00, max_container_radiation: 5.00
  phases: pid->name:capacity,count,radiation
    p1->packing:4,0,0.00,{glass,metal,plastic,liquid}
    p2->crushing:3,0,0.00,{glass,metal,plastic}
  containers: cid->pid->material,radioactivity
->new_container("c",[glass, 3],"p1")
  state 4 ok
  max_phase_radiation: 10.00, max_container_radiation: 5.00
  phases: pid->name:capacity,count,radiation
    p1->packing:4,1,3.00,{glass,metal,plastic,liquid}
    p2->crushing:3,0,0.00,{glass,metal,plastic}
  containers: cid->pid->material,radioactivity
    c->p1->glass,3.00
->new_container("c2",[plastic, 4],"p1")
  state 5 ok
  max_phase_radiation: 10.00, max_container_radiation: 5.00
  phases: pid->name:capacity,count,radiation
    p1->packing:4,2,7.00,{glass,metal,plastic,liquid}
    p2->crushing:3,0,0.00,{glass,metal,plastic}
  containers: cid->pid->material,radioactivity
    c->p1->glass,3.00
    c2->p1->plastic,4.00
->new_container("c3",[metal, 3],"p2")
  state 6 ok
  max_phase_radiation: 10.00, max_container_radiation: 5.00
  phases: pid->name:capacity,count,radiation
    p1->packing:4,2,7.00,{glass,metal,plastic,liquid}
    p2->crushing:3,1,3.00,{glass,metal,plastic}
  containers: cid->pid->material,radioactivity
    c->p1->glass,3.00
    c2->p1->plastic,4.00
    c3->p2->metal,3.00
->new_container("c4",[liquid, 3],"p1")
  state 7 ok
  max_phase_radiation: 10.00, max_container_radiation: 5.00
  phases: pid->name:capacity,count,radiation
    p1->packing:4,3,10.00,{glass,metal,plastic,liquid}
    p2->crushing:3,1,3.00,{glass,metal,plastic}
  containers: cid->pid->material,radioactivity
    c->p1->glass,3.00
    c2->p1->plastic,4.00
    c3->p2->metal,3.00
    c4->p1->liquid,3.00
->move_container("c","p1","p2")
  state 8 ok
  max_phase_radiation: 10.00, max_container_radiation: 5.00
  phases: pid->name:capacity,count,radiation
    p1->packing:4,2,7.00,{glass,metal,plastic,liquid}
    p2->crushing:3,2,6.00,{glass,metal,plastic}
  containers: cid->pid->material,radioactivity
    c->p2->glass,3.00
    c2->p1->plastic,4.00
    c3->p2->metal,3.00
    c4->p1->liquid,3.00
